runtime/queries/metricsview_aggregation.go

```diff
@@ -579,32 +584,127 @@ func (q *MetricsViewAggregation) buildMetricsAggregationSQL(mv *runtimev1.Metric
 			limitClause,                     // 8
 		)
 	} else {
-		sql = fmt.Sprintf("SELECT %s FROM %s %s %s %s %s %s %s OFFSET %d",
-			strings.Join(selectCols, ", "),
-			safeName(mv.Table),
-			strings.Join(unnestClauses, ""),
-			whereClause,
-			groupClause,
-			havingClause,
-			orderClause,
-			limitClause,
-			q.Offset,
-		)
-	}
+		/*
+			Example:
+			SELECT t.d1, t.d2, t.d3, t2.m1 (SELECT d1, d2, d3, m1 FROM t WHERE ...  GROUP BY d1, d2, d3 HAVING m1 > 10 ) t LEFT JOIN (
+				SELECT d1, d2, d3, m1 FROM t WHERE ... AND (d4 = 'Safari') GROUP BY d1, d2, d3 HAVING m1 > 10
+			)  t2 ON (COALESCE(t.d1, 'val') = COALESCE(t2.d1, 'val') and COALESCE(t.d2, 'val') = COALESCE(t2.d2, 'val') and ...)
+			WHERE t2.m1 > 10
+			ORDER BY ...
+			LIMIT 100
+			OFFSET 0
+
+			This JOIN mirrors functionality of SELECT d1, d2, d3, m1 FILTER (WHERE d4 = 'Safari') FROM t WHERE... GROUP BY d1, d2, d3
+			bacause FILTER cannot be applied for arbitrary measure, ie sum(a)/1000
+		*/
+		if filterCount == 1 {
```

Comment:
Can we pull this case into a separate function? The level of nesting and complexity here makes it hard to follow

Reply:
it will be long signature function